### PR TITLE
4000 - Dashboard API

### PR DESCRIPTION
**Pull request by sorja**

### CODE REVIEW

**Comment by sorja** [comment]
@minotogna Opened PR for basis of discussion, let me know how the api should look.

Currently:

```
curl localhost:9001/api/data-query?assessmentName=fra&cycleName=2025
```

Returns an `DashboardItem<DashboardItemType>`

**Comment by minotogna** [comment]
> @minotogna Opened PR for basis of discussion, let me know how the api should look.
> 
> 
> 
> Currently:
> 
> 
> 
> ```
> 
> curl localhost:9001/api/data-query?assessmentName=fra&cycleName=2025
> 
> ```
> 
> 
> 
> Returns an `DashboardItem<DashboardItemType>`

APIs should be cycle based .
For now the response is ok .

**Comment by sorja** [comment]
> > @minotogna Opened PR for basis of discussion, let me know how the api should look.
> > Currently:
> > ```
> > 
> > curl localhost:9001/api/data-query?assessmentName=fra&cycleName=2025
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Returns an `DashboardItem<DashboardItemType>`
> 
> APIs should be cycle based . For now the response is ok .

If I understood correctly:
Api under `api/cycleData` and controller under `cycleData` and repository under `assessmentCycle`

Sure

**Comment by minotogna** [comment]
> > > @minotogna Opened PR for basis of discussion, let me know how the api should look.
> 
> > > Currently:
> 
> > > ```
> 
> > > 
> 
> > > curl localhost:9001/api/data-query?assessmentName=fra&cycleName=2025
> 
> > > ```
> 
> > > 
> 
> > > 
> 
> > >     
> 
> > >       
> 
> > >     
> 
> > > 
> 
> > >       
> 
> > >     
> 
> > > 
> 
> > >     
> 
> > >   
> 
> > > Returns an `DashboardItem<DashboardItemType>`
> 
> > 
> 
> > APIs should be cycle based . For now the response is ok .
> 
> 
> 
> If I understood correctly:
> 
> Api under `api/cycleData` and controller under `cycleData` and repository under `assessmentCycle`
> 
> 
> 
> Sure

Don't we have the repository nod ext already ?

File: src/server/api/cycleData/dashboard/getDashboardItems.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Response } from 'express'
+
+import { CycleRequest } from 'meta/api/request'
+
+import { AssessmentController } from 'server/controller/assessment'
+import { DashboardController } from 'server/controller/cycleData/dashboard'
+import Requests from 'server/utils/requests'
+
+type Query = { query: string; limit: string }
```
Comment (minotogna) [inline src/server/api/cycleData/dashboard/getDashboardItems.ts]: where do u use this ?

**Review by minotogna — COMMENTED** [review]
👍 just a minor comment .

File: src/server/api/cycleData/dashboard/getDashboardItems.ts (hunk 1 above)
Comment (sorja) [reply src/server/api/cycleData/dashboard/getDashboardItems.ts]: good catch, typo/missed

**Review by minotogna — APPROVED** [review]

**Pull request merged**